Update ora/xml_util_pkg.pkb to fix #27

datRedHeadedGuy opened the pull request:
Modified the xml_util_pkg and changed the use of XMLType.extract(xpath).getStringVal() to EXTRACTVALUE(XMLType, xpath) so that special characters are not escaped.

datRedHeadedGuy committed:
Update ora/xml_util_pkg.pkb

Modified the xml_util_pkg and changed the use of XMLType.extract(xpath).getStringVal() to EXTRACTVALUE(XMLType, xpath) so that special characters are not escaped.

mortenbra commented:
Hesitant to approve this, as the EXTRACTVALUE function has been deprecated since at least Oracle 11.2 (see https://docs.oracle.com/cd/E11882_01/server.112/e41084/functions061.htm#SQLRF06173 ).

Can you describe the problem in more detail?

eaolson commented:
Example:

```
declare
    x xmltype;
    ex varchar2(100);
begin
    x := xmltype ('<root><foo bar="three &amp; four">one &amp; two</foo></root>');
    ex := xml_util_pkg.extract_value( x, '/root/foo/text()' );
    dbms_output.put_line( ex );
    ex := xml_util_pkg.extract_value( x, '/root/foo/@bar' );
    dbms_output.put_line( ex );
end;
```

Returns:
```
one &amp; two
three &amp; four
```

Wouldn't the expected result have the &amp; unescaped to just "&"? Would calling dbms_xmlgen.convert() be a better solution?

datRedHeadedGuy committed:
Merge branch 'mortenbra:master' into ooxml_util_pkg-XLSX-functions-return-XML-encoded-strings-#27

datRedHeadedGuy committed:
Update xml_util_pkg.pkb

Removed original changes and replaced them with a new package function, get_unescaped_string().

datRedHeadedGuy committed:
Update xml_util_pkg.pks

Added new package function, get_unescaped_string().

datRedHeadedGuy commented:
I removed my changes where I replaced XMLType.extract(xpath).getStringVal() with the EXTRACTVALUE function and just pass the string value into a new function, get_unescaped_string(), before returning it.

datRedHeadedGuy commented:
resolves #27

datRedHeadedGuy committed:
Update xml_util_pkg.pkb

datRedHeadedGuy committed:
Update xml_util_pkg.pks
